src/main/java/org/radarcns/util/EmailSenders.java

```diff
@@ -0,0 +1,85 @@
+package org.radarcns.util;
+
+import org.radarcns.config.MonitorConfig;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Class to store {@link EmailSender} associated with each project.
+ */
+public class EmailSenders {
+
+    private Map<String,EmailSender> emailSenderMap;
+    private IOException exception;
+
+    /**
+     * Default Constructor
+     */
+    public EmailSenders() {
+        emailSenderMap = new HashMap<>();
+    }
+
+    /**
+     * Initialize the {@link EmailSender} map using the configuration.
+     *
+     * @param  config  Configuration of the Monitor containing project
+     *                 and email address mapping
+     * @throws IOException
+     */
+    public EmailSenders(MonitorConfig config) throws IOException{
+        emailSenderMap = new HashMap<>();
+        addSender(config);
+    }
+
+    /**
+     *
+     * Parses the {@link MonitorConfig} to map the corresponding
+     * {@link EmailSender} to each project. A project can have a list of
+     * associated email addresses.
+     *
+     * @param  config  Configuration of the Monitor containing project
+     *                 and email address mapping
+     * @throws IOException
+     */
+
+    private void addSender(MonitorConfig config) throws IOException{
+        exception = null;
+        config.getNotifyConfig().stream().forEach(s -> {
```

Comment:
why not use
```java
protected Map<String, EmailSender> createSenders(MonitorConfig config) {
return config.getNotifyConfig().stream()
    .collect(Collectors.toMap(NotifyConfig::getProjectId, s-> {
        try {
           return new EmailSender(config.getEmailHost(), config.getEmailPort(),
               config.getEmailUser(), s.getEmailAddress());
        } catch (IOException ex) {
            throw new IllegalArgumentException("Cannot create email sender", ex);
        }
    });
}
```
and remove the exception from the class?

Or even just make parsing the MonitorConfig static:
```java
public static EmailSenders parseConfig(MonitorConfig config) {
return new EmailSenders(config.getNotifyConfig().stream()
    .collect(Collectors.toMap(NotifyConfig::getProjectId, s-> {
        try {
           return new EmailSender(config.getEmailHost(), config.getEmailPort(),
               config.getEmailUser(), s.getEmailAddress());
        } catch (IOException ex) {
            throw new IllegalArgumentException("Cannot create email sender", ex);
        }
    }));
}
```
If you need the map to be modifiable, you could always wrap this in a `new HashMap`.

Reply:
The reason for throwing the IOException here was because one of the test expected an IOException if the email sender was created without a proper config. Should I update the test as well - https://github.com/RADAR-base/RADAR-Backend/blob/dc162671257c80791c04b5b2e6f4518b204ad636/src/test/java/org/radarcns/monitor/KafkaMonitorFactoryTest.java#L63 ?

Reply:
ok, good point. How about not using streams then?
```java
map = new HashMap<>();
for (NotifyConfig notifyConfig : config.getNotifyConfig()) {
    map.put(notifyConfig.getProjectId(), new EmailSender(...));
}
return map;
```
In any case, `forEach` is not meant to modify an existing non-concurrent map.